src/fastoad/gui/analysis_and_plots.py

```diff
@@ -351,58 +352,136 @@ def mass_breakdown_sun_plot(
                            default format will be assumed.
     :param input_mass_name: the variable name for the mass input as defined in the mission
                             definition file.
+    :param mission_name: the name of the specific mission for wich the mass breakdown is plotted.
+                         If not provided, the sizing mission configuration is plotted.
     :return: sunburst plot figure
     """
     variables = VariableIO(aircraft_file_path, file_formatter).read()
 
-    var_names_and_new_units = {
-        input_mass_name: "kg",
-        "data:weight:aircraft:OWE": "kg",
-        "data:weight:aircraft:payload": "kg",
-        "data:weight:aircraft:sizing_onboard_fuel_at_input_weight": "kg",
-    }
+    # Check if the provided mission_name is an existing mission
+    missions_set = []
+    for v in variables:
+        if v.name.startswith("data:mission:"):
+            missions_set.append(v.name.split(":")[2])
+    missions_set = set(missions_set)  # set of all the missions defined in aircraft_file_path
+
+    if mission_name:  # Check if mission_name is used and exists
+        if mission_name not in missions_set:
+            raise ValueError(
+                f"The provided mission_name {mission_name} does not correspond to an existing"
+                + "mission. The available mission(s) are: {missions_set}."
+            )
+
+    if not mission_name:  # Print sizing mission
+        var_names_and_new_units = {
+            input_mass_name: "kg",
+            "data:weight:aircraft:OWE": "kg",
+            "data:weight:aircraft:payload": "kg",
+            "data:weight:aircraft:sizing_onboard_fuel_at_input_weight": "kg",
+        }
+        tow, owe, payload, onboard_fuel_at_takeoff = _get_variable_values_with_new_units(
+            variables, var_names_and_new_units
+        )
+    else:
+        payload_test_name = f"data:mission:{mission_name}:payload"
+        if payload_test_name not in variables.names():
+            payload_test_name = "data:weight:aircraft:payload"
```

Comment:
Good to have taken this case into account.
You should test that by using also `MTOW_mission` in the unit test.